weldx/transformations/local_cs.py

```diff
@@ -664,22 +662,18 @@ def as_rotation(self) -> Rot:  # pragma: no cover
         """
         return Rot.from_matrix(self.orientation.values)
 
-    def _interp_time_orientation(self, time: Time) -> Union[xr.DataArray, np.ndarray]:
+    def _interp_time_orientation(self, time: Time) -> xr.DataArray:
         """Interpolate the orientation in time."""
-        if self.orientation.ndim == 2:  # don't interpolate static
-            orientation = self.orientation
-        elif time.max() <= self.time.min():  # only use edge timestamp
-            orientation = self.orientation.values[0]
-        elif time.min() >= self.time.max():  # only use edge timestamp
-            orientation = self.orientation.values[-1]
-        else:  # full interpolation with overlapping times
-            orientation = ut.xr_interp_orientation_in_time(self.orientation, time)
-
-        if len(orientation) == 1:  # remove "time dimension" for single value cases
-            return orientation[0].values
-        return orientation
-
-    def _interp_time_coordinates(self, time: Time) -> Union[xr.DataArray, np.ndarray]:
+        if "time" not in self.orientation.dims:  # don't interpolate static
```

Comment:
this is probably also a safer way to check for 'static' systems

Reply:
True, but also more expensive. As already laid out in my other comments, we are dealing with a predefined internal structure that will most likely never change. So basically, we are just trading a bit of performance for a bit of syntactic sugar. For safety, we have our tests. I see no need for generalizations on a specialized internal data structure.